src/main/java/edu/ucsb/cs156/example/services/UCSBSubjectsService.java

```diff
@@ -1,57 +1,71 @@
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+
+import org.springframework.beans.factory.annotation.Value;
+
 @Slf4j
 @Service("UCSBSubjects")
 public class UCSBSubjectsService {
 
-    public List<UCSBSubject> get() {
-
-        // TODO: Replace with a service that will call the UCSB
-        // developer API at the endpoint
-        // https://api.ucsb.edu/students/lookups/v1/subjects?includeInactive=false
-        // convert the JSON to a List<UCSBSubjects> object
-        // return that object.
-
-        UCSBSubject us1 = UCSBSubject.builder()
-                .subjectCode("ANTH")
-                .subjectTranslation("Anthropology")
-                .deptCode("ANTH")
-                .collegeCode("L&S")
-                .relatedDeptCode(null)
-                .inactive(false)
-                .build();
-
-        UCSBSubject us2 = UCSBSubject.builder()
-                .subjectCode("ART  CS")
-                .subjectTranslation("Art (Creative Studies)")
-                .deptCode("CRSTU")
-                .collegeCode("CRST")
-                .relatedDeptCode(null)
-                .inactive(false)
-                .build();
-
-        UCSBSubject us3 = UCSBSubject.builder()
-                .subjectCode("CH E")
-                .subjectTranslation("Chemical Engineering")
-                .deptCode("CNENG")
-                .collegeCode("ENGR")
-                .relatedDeptCode(null)
-                .inactive(false)
-                .build();
-
-        List<UCSBSubject> temporaryFakeList = new ArrayList<>();
-        temporaryFakeList.addAll(Arrays.asList(us1, us2, us3));
-        log.info("temporaryFakeList={}",temporaryFakeList);
-        return temporaryFakeList;
-    }
-
-}
+        @Value("${app.ucsb.api.key}") private String ucsbApiKey;
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        private RestTemplate restTemplate = new RestTemplate();
+
+        public UCSBSubjectsService(RestTemplateBuilder restTemplateBuilder) {
+                restTemplate = restTemplateBuilder.build();
+        }
+
+        public static final String ENDPOINT = "https://api.ucsb.edu/students/lookups/v1/subjects?includeInactive=false";
+        
+        public String getJSON() throws HttpClientErrorException {
+                HttpHeaders headers = new HttpHeaders();
+                headers.setAccept(List.of(MediaType.APPLICATION_JSON));
+                headers.setContentType(MediaType.APPLICATION_JSON);
+                headers.set("ucsb-api-key", this.ucsbApiKey);
+                headers.set("ucsb-api-version", "1.6");
+
+                HttpEntity<String> entity = new HttpEntity<>(headers);
+
+                Map<String, String> uriVariables = Map.of("subjectCode", "", "subjectTranslation", "");
```

Comment:
Providing these is not required.

Reply:
Good catch. Fixing this in the next commit.